js/battle.js

```diff
@@ -5317,7 +5317,8 @@ var Battle = (function () {
 		case 'chat':
 		case 'c':
 			name = args[1];
-			if (this.ignoreSpects && !this.getSide(name).battle) break;
+			if ((Tools.prefs('ignorespectsforever') || this.ignoreSpects) && !this.getSide(toId(name)).battle) break;
```

Comment:
Instead of calling <code>Tools.prefs('ignorespectsforever')</code> for every chat message, you could store it in <code>this.ignoreSpects</code> when initializing the battle.